src/System.IO.FileSystem/tests/Performance/Perf.Directory.cs

```diff
@@ -55,5 +57,79 @@ public void Exists()
             // Teardown
             Directory.Delete(testFile);
         }
+
+        [Benchmark]
+        [InlineData(10)]
+        [InlineData(100)]
+        [InlineData(1000)]
+        [OuterLoop("Takes a lot of time to finish")]
+        public void RecursiveCreateDirectory(int depth)
+        {
+            if (!PlatformDetection.IsWindows && depth == 1000)
+                return;
+            
+            // Setup
+            string rootDirectory = GetTestFilePath();
+            StringBuilder sb = new StringBuilder(5000);
+            for (int i = 0; i < depth; i++)
+            {
+                sb.Append(@"\a");
+            }
+
+            string path = sb.ToString();
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < 1000; i++)
```

Comment:
@JeremyKuhne how to decide the number ```1000```. with 20000 I am getting ```nan``` as output

Reply:
Whatever runs in a measurable amount of time. You don't need to run forever- just something that gives a large enough time to filter out some of the noise (few hundred ms or larger).  You might want to consider passing in a different count for each of your data points.